### PR TITLE
Init 1.12 development

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by jschueller** [comment]
We did not deprecate FieldFunction::getSpatialDimension, should it be renamed to getMeshDimension or something ?

**Comment by sofianehaddad** [comment]
I think we could avoid 91868b9 and afda761 as SecondOrderModel class will be removed.
Otherwise ok for me

File: python/src/SecondOrderModel_doc.i.in [hunk 1]
```diff
@@ -4,18 +4,18 @@
 OT_SecondOrderModel_doc
```
Comment (sofianehaddad) [inline python/src/SecondOrderModel_doc.i.in]: Class will be removed so no needed.

File: python/src/SecondOrderModelImplementation_doc.i.in [hunk 2]
```diff
@@ -27,7 +27,7 @@ OT_SecondOrderModel_doc
 
 // ---------------------------------------------------------------------
```
Comment (sofianehaddad) [inline python/src/SecondOrderModelImplementation_doc.i.in]: Second order models will be removed. It is useless

**Pull request merged**